### PR TITLE
Gracefully fall back to unlit shading in ModelExperimental for models without normals

### PR DESCRIPTION
**Pull request by ptrgags**
This PR makes two changes to make rendering models handle edge cases more gracefully:

1. If a primitive in a `ModelExperimental` does not have normals, now the lighting mode falls back gracefully to UNLIT, since this looks better than attempting to render PBR with default settings (which often results in everything turning solid black). Technically speaking, this behavior is not correct by the glTF spec, but in Cesium the unlit behavior is consistent with the old `Model`. Unlit shading is very common in geospatial data (e.g. photogrammetry) where we want to render the textures directly. 
2. For `CustomShader`, I noticed that if you specify the `lightingModel` or `isTranslucent` but do not include `vertexShaderText`/`fragmentShaderText`, these settings were being optimized out. I moved the if checks earlier in `CustomShaderPipelineStage` so they always get applied, this is more useful.

[Local Sandcastle for testing](http://localhost:8080/Apps/Sandcastle/index.html#c=tVVdb9s2FP0rhJ7sIaYkW7Ed1w3WpU22ocmK2tiATXtgpGuLCEUKJOU4G/LfdylKipq62F7mB4kf536de66cKWksOXB4BE3eEgmP5AoMr0v6a3M2SoOs2V8paRmXoNNg/CaVqfQ2NBMqe6BZrTVIu+UloJfWw8+14Ey+ZxboTqvyJ6OW8ygepZKQNJhG03gSx5PoYjuNVtPFaragUTyP5kmyOJ8ly/kyuZgtfk+DVPp4YUhuVQ7CpDJrki6bHYb7w3n82z0IsXC0K3T/gzqS0Z0id0qXDFETYgpVi5zcA6ml4HacBmfeBJ2BgAzNdrXMLFeSjMadP0L8ZRN61J25AsJNBZkJsTwW+sTCG2F3HxXLQYcY/061wcO92F5/cUL3iOwTcL+oW7pi3fu5ufTPU7VdK00yVVZMc6Pk/1ENhpm4iiZTt/zvKafyz6ZhvktF9UpXPwLLudx/4jYrPivh02jvbpktqFWfEcGkGcXLaNz4jPzTS8H73fEj5NealbDViN0hsS/K648MRXky4XHqure5AVQys0q3apRK26KtLg0ewdgX4fU8DrmrtTgjBfB9YVt2fVqVMrwB97lcMW1xxeSsmYL3sNcApiV/Ek9nNFokyTy+aKlNEhqdR7NFNG8PfBS39vmQdlypybAKWmleYsgDGKqhVAd4h5z6lgzmBPP5lhXL8zabrgvO4MOxAsTgUDPRJO6kMOp15NSwIo6F7qQJgw3U/Lg60YfiVduH3RgoseNvoDNU0GDXev4gBK+M4jn97WazTAaAE8LohdrCnsdDNpvEkTuWP33COrkBaguQo5f5aRD9EHUfPoygGepLPbyzfQVf5f8t5TO5H9Z9egCS8/HZv2Em8RcgX829qqWLtakK0EA1YmtDviPnPRV+4XXy7F7Pjo0Nk3nGjBXgZLFVStwzfQuy9hyY8ZvgLFgb+yTgsnP1PS8rHB8nhhGloYWyEvjNN+F9nT2ApZkx3RdiHQ5N1zk/EJ6/PfEXQzLBjMGbXS3Ehv8FaXC5DhH/lalQDam/HEAL9uRgRXz50R9SStchbk9bWl/dK8//AA)

@j9liu could you review when you get a chance?

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @ptrgags!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/gracefully-unlit/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

File: CHANGES.md [hunk 1]
```diff
@@ -26,6 +26,7 @@
 - Fixed an issue with Firefox and dimensionless SVG images. [#9191](https://github.com/CesiumGS/cesium/pull/9191)
 - Fixed `ShadowMap` documentation for `options.pointLightRadius` type. [#10195](https://github.com/CesiumGS/cesium/pull/10195)
 - Fixed evaluation of `minimumLevel` on metadataFailure for TileMapServiceImageryProvider.
+- Fixed a bug where models without normals would render as solid black. Now such models will fall back to unlit shading. [#10237](https://github.com/CesiumGS/cesium/pull/10237)
```
Comment (j9liu) [inline CHANGES.md:29]:
```suggestion
- Fixed a bug where models without normals would render as solid black. Now such models will use unlit shading. [#10237](https://github.com/CesiumGS/cesium/pull/10237)
```

Comment (j9liu) [reply CHANGES.md:29]: I think "fall back to" is a bit unusual phrasing, just say "use"

File: Specs/Data/Models/GltfLoader/BoxNoNormals/README.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# A Box Without Normals
+
+This model is a variation on Box-Gltf-2 except with the `NORMAL` attribute
+is removed. No `KHR_materials_unlit` is added. This is intentional to test
+fallback behavior to unlit shading.
```
Comment (j9liu) [inline Specs/Data/Models/GltfLoader/BoxNoNormals/README.md:5]:
```suggestion
# A Box Without Normals

This model is a variation on Box-Gltf-2 except the `NORMAL` attribute has been
removed. `KHR_materials_unlit` has not been added. This is to intentionally test
the fallback behavior when a model without normals is loaded in -- it should
instead use unlit shading.
```

**Comment by j9liu** [comment]
@ptrgags these changes look good, and the Sandcastle works as intended. I made a few suggestions unrelated to the code -- once those get addressed I'll go ahead with the merge!

**Comment by ptrgags** [comment]
@j9liu updated!

**Comment by j9liu** [comment]
Merging now. Thanks @ptrgags !

**Pull request merged**